src/os/bluestore/BlueStore.cc

```diff
@@ -5770,7 +5770,9 @@ void BlueStore::_do_write_small(
 	  bdev->aio_write(offset, t,
 			  &txc->ioc, wctx->buffered);
 	});
-      b->blob.calc_csum(b_off, padded);
+      if (b->blob.csum_type) {
```

Comment:
IMHO no much sense in this fix as calc_csum performs verification by itself - see corresponding switch inside...